src/runtime/entrypoints/main.ts

```diff
@@ -323,7 +326,12 @@ function _walkInner(
         };
         for (let i = 0; i < sib.attributes.length; i++) {
           const attr = sib.attributes[i];
-          props[attr.nodeName] = attr.nodeValue;
+
+          // Boolean attributes are always `true` when present.
+          // See: https://developer.mozilla.org/en-US/docs/Glossary/Boolean/HTML
+          props[attr.nodeName] = BOOLEAN_ATTR.test(attr.nodeName)
+            ? true
+            : attr.nodeValue;
```

Comment:
Have you benched if regex is faster than searching in an array?

Reply:
Good call, did a quick benchmark and given that we're dealing with short arrays it's faster. Thought about this some more if there is a way not to ship those strings at all to the browser and found an even better approach. Since we already _have_ all attribute names we care about and they all exist as properties on DOM nodes we can do a simple `typeof value === "boolean"` check.